client/scala/src/main/scala/ai/verta/client/Client.scala

```diff
@@ -47,11 +51,65 @@ class Client(conn: ClientConnection) {
 
   def getExperimentRun(id: String)(implicit ec: ExecutionContext) = {
     clientSet.experimentRunService.getExperimentRunById(Some(id))
-      .flatMap(r => if (r.experiment_run.isEmpty) Success(null) else Try[ExperimentRun]({
+      .flatMap(r => Try[ExperimentRun]({
         getExperiment(r.experiment_run.get.experiment_id.get) match {
           case Success(expt) => new ExperimentRun(clientSet, expt, r.experiment_run.get)
           case Failure(x) => throw x
         }
       }))
   }
+
+  /** Get the repository by name (and workspace). If not exist, create new repository
+   * @param name Name of the Repository
+   * @param workspace Workspace under which the Repository with name name exists. If not provided, the current user’s personal workspace will be used.
+   */
+  def getOrCreateRepository(name: String, workspace: Option[String] = None)(implicit ec: ExecutionContext) = {
+    GetOrCreateEntity.getOrCreate[Repository](
+      get = () => {
+        clientSet.versioningService.GetRepository(
+          id_named_id_workspace_name = if (workspace.isDefined) workspace.get else getPersonalWorkspace(),
+          id_named_id_name = urlEncode(name)
+        )
+        .map(r => new Repository(clientSet, r.repository.get))
+      },
+      create = () => {
+        clientSet.versioningService.CreateRepository(
+          id_named_id_workspace_name = if (workspace.isDefined) workspace.get else getPersonalWorkspace(),
+          body = VersioningRepository(
+            name = Some(name),
+            workspace_id = workspace
+          )
+        ).map(r => new Repository(clientSet, r.repository.get)
+        )
+      }
+    )
+  }
+
+  /** Get repository based on id
+   *  @param id id of the repository
+   *  @return the repository
+   */
+  def getRepository(id: String)(implicit ec: ExecutionContext): Try[Repository] = {
+    clientSet.versioningService.GetRepository2(
+      id_repo_id = BigInt(id)
+    ).map(r => new Repository(clientSet, r.repository.get))
+  }
+
+
+  /** Get repository based on id
+   *  @param id id of the repository
+   */
+  def deleteRepository(id: String)(implicit ec: ExecutionContext): Try[VersioningDeleteRepositoryRequestResponse] = {
```

Comment:
Don't return the protobuf object `VersioningDeleteRepositoryRequestResponse`. Return something more meaningful to the user

Reply:
Hmm for this function, the only piece of meaningful information that user should retrieve is whether the delete operation succeeds or not. VersioningDeleteRepositoryRequestResponse is empty, so I don't really know what to map it too. Or should I just return an "empty" Success object?

Reply:
`Try[Unit]`.

Reply:
Oh ok. I'll make modifications to this and the similar below.

Reply:
Resolved. But do check if there's a more scala-ic way to do this. I followed one of your example with pattern matching, but I'm pretty sure map(_ => ()) do the same thing and is a bit shorter.